src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -55,13 +62,42 @@ internal bool AddGas(long gas)
             return testMode || GasConsumed <= gas_amount;
         }
 
+        internal void CallFromNativeContract(Action onComplete, UInt160 hash, string method, params StackItem[] args)
+        {
+            invocationStates.Add(CurrentContext, new InvocationState
+            {
+                ReturnType = typeof(void),
+                Callback = onComplete
+            });
+            CallContract(hash, ContractParameterType.Void, method, new VMArray(args));
+        }
+
+        internal void CallFromNativeContract<T>(Action<T> onComplete, UInt160 hash, string method, params StackItem[] args)
+        {
+            invocationStates.Add(CurrentContext, new InvocationState
```

Comment:
Should replace `CurrentContext` by the context of `CallContract` in L82? `CurrentContext` is the native contract, is not the `context` of user contract, which will be called.

Reply:
No, it shouldn't. We use the `CurrentContext` as key.